nc_test4/tst_varsperf.c

```diff
@@ -0,0 +1,149 @@
+/*
+Create a netcdf-4 file with
+a large variable for the purpose
+of doing performance test on the
+new NC4_get/put_vars functions.
+
+WARNING: do not attempt to run this
+under windows because of the use
+of gettimeofday().
+
+*/
+
+#include "config.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/time.h>
+#include <assert.h>
+#include "netcdf.h"
+
+
+#define FILE "bigvars.nc"
+#define VAR "bigvar"
+#define NDIMS 2
+#define DIM0 "d0"
+#define DIM1 "d1"
+#define DIMSIZE0 1024
+#define DIMSIZE1 1024
+#define TOTALSIZE (DIMSIZE0*DIMSIZE1)
+
+#define CHECK(expr) assert((expr) == NC_NOERR)
```

Comment:
Unfortunately we cannot use asserts in tests like this.

The reason is that this will cause the test to fail to compile with -DNDEBUG builds. The -DNDEBUG is usual in HPC builds (and also in code coverage builds, which is how my CI system caught this.)

So I have made minor modifications to the test to no longer use asserts, but instead to use the macros in err_macros.h.